test flake: capacity

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_external-provisioner/605/pull-kubernetes-csi-external-provisioner-unit/1381723395330150400
```
--- FAIL: TestCapacityController (0.00s)
    --- FAIL: TestCapacityController/reuse_one_capacity_object,_update_capacity (0.10s)
        capacity_test.go:1135: unexpected CSIStorageCapacity v1beta1.CSIStorageCapacity{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"csisc-13", GenerateName:"", Namespace:"testns", SelfLink:"", UID:"test-capacity-2", ResourceVersion:"CSISC-REV-1", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"csi.storage.k8s.io/drivername":"test-driver", "csi.storage.k8s.io/managed-by":"external-provisioner"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference{v1.OwnerReference{APIVersion:"apps/v1", Kind:"statefulset", Name:"test-driver", UID:"309cd460-2d62-4f40-bbcf-b7765aac5a6d", Controller:(*bool)(0xc00059e640), BlockOwnerDeletion:(*bool)(nil)}}, Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, NodeTopology:(*v1.LabelSelector)(0xc0001d3ce0), StorageClassName:"triple-sc", Capacity:(*resource.Quantity)(0xc0006eedc0), MaximumVolumeSize:(*resource.Quantity)(nil)}
            unexpected CSIStorageCapacity v1beta1.CSIStorageCapacity{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"csisc-14", GenerateName:"", Namespace:"testns", SelfLink:"", UID:"test-capacity-3", ResourceVersion:"CSISC-REV-1", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"csi.storage.k8s.io/drivername":"test-driver", "csi.storage.k8s.io/managed-by":"external-provisioner"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference{v1.OwnerReference{APIVersion:"apps/v1", Kind:"statefulset", Name:"test-driver", UID:"309cd460-2d62-4f40-bbcf-b7765aac5a6d", Controller:(*bool)(0xc00059e641), BlockOwnerDeletion:(*bool)(nil)}}, Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, NodeTopology:(*v1.LabelSelector)(0xc0001d3d00), StorageClassName:"direct-sc", Capacity:(*resource.Quantity)(0xc0006eee00), MaximumVolumeSize:(*resource.Quantity)(nil)}
            unexpected CSIStorageCapacity v1beta1.CSIStorageCapacity{TypeMeta:v1.TypeMeta{Kind:"", APIVersion:""}, ObjectMeta:v1.ObjectMeta{Name:"csisc-9", GenerateName:"", Namespace:"testns", SelfLink:"", UID:"test-capacity-1", ResourceVersion:"CSISC-REV-1", Generation:0, CreationTimestamp:v1.Time{Time:time.Time{wall:0x0, ext:0, loc:(*time.Location)(nil)}}, DeletionTimestamp:(*v1.Time)(nil), DeletionGracePeriodSeconds:(*int64)(nil), Labels:map[string]string{"csi.storage.k8s.io/drivername":"test-driver", "csi.storage.k8s.io/managed-by":"external-provisioner"}, Annotations:map[string]string(nil), OwnerReferences:[]v1.OwnerReference{v1.OwnerReference{APIVersion:"apps/v1", Kind:"statefulset", Name:"test-driver", UID:"309cd460-2d62-4f40-bbcf-b7765aac5a6d", Controller:(*bool)(0xc00059e643), BlockOwnerDeletion:(*bool)(nil)}}, Finalizers:[]string(nil), ClusterName:"", ManagedFields:[]v1.ManagedFieldsEntry(nil)}, NodeTopology:(*v1.LabelSelector)(0xc0001d3d40), StorageClassName:"direct-sc", Capacity:(*resource.Quantity)(0xc0006eee80), MaximumVolumeSize:(*resource.Quantity)(nil)}
            expected CSIStorageCapacity {uid:test-capacity-4 resourceVersion:CSISC-REV-1 segment:[{Key:layer0 Value:bar}] storageClassName:triple-sc quantity:6Gi maxVolume: owner:<nil> managedByID:} not found
            expected CSIStorageCapacity {uid:test-capacity-4 resourceVersion:CSISC-REV-1 segment:[{Key:layer0 Value:bar}] storageClassName:triple-sc quantity:6Gi maxVolume: owner:<nil> managedByID:} not found
            expected CSIStorageCapacity {uid:test-capacity-4 resourceVersion:CSISC-REV-1 segment:[{Key:layer0 Value:bar}] storageClassName:triple-sc quantity:6Gi maxVolume: owner:<nil> managedByID:} not found
```


https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-csi_external-provisioner/605/pull-kubernetes-csi-external-provisioner-unit/1381726066699145216

```
 --- FAIL: TestCapacityController (0.00s)
    --- FAIL: TestCapacityController/no_owner (0.10s)
        capacity_test.go:1128: metrics after prepare: expected goal/current/obsolete object numbers 1/0/0: 
            metric output does not match expectation; want:
            
            # HELP csistoragecapacities_desired_current [ALPHA] Number of CSIStorageCapacity objects that exist and are supposed to be managed automatically.
            # TYPE csistoragecapacities_desired_current gauge
            # TYPE csistoragecapacities_desired_goal gauge
            csistoragecapacities_desired_goal 0
            # HELP csistoragecapacities_obsolete [ALPHA] Number of CSIStorageCapacity objects that exist and will be deleted automatically. Objects that exist and may need an update are not considered obsolete and therefore not included in this value.
            # TYPE csistoragecapacities_obsolete gauge
            csistoragecapacities_obsolete 0 
```

--- a/pkg/capacity/capacity_test.go
+++ b/pkg/capacity/capacity_test.go
@@ -1089,6 +1089,7 @@ func TestCapacityController(t *testing.T) {
 	}
 
 	for name, tc := range testcases {
+		tc := tc
 		t.Run(name, func(t *testing.T) {
 			// Running in parallel is possible because only logging uses a global instance.
 			t.Parallel()
@@ -1766,6 +1767,7 @@ func TestTermToSegment(t *testing.T) {
 	}
 
 	for name, tc := range testcases {
+		tc := tc
 		t.Run(name, func(t *testing.T) {
 			t.Parallel()
 
@@ -1920,6 +1922,7 @@ func TestRefresh(t *testing.T) {
 	}
 
 	for name, tc := range testcases {
+		tc := tc
 		t.Run(name, func(t *testing.T) {
 			// Running in parallel is possible because only logging uses a global instance.
 			t.Parallel()

